repos/fdbt-site/src/pages/products/services.tsx

```diff
@@ -80,10 +89,53 @@ const getTag = (startDate: string, endDate: string) => {
     }
 };
 
-export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: ServicesProps }> => {
-    const services = await getBodsServicesByNoc(getAndValidateNoc(ctx));
+export const showProductAgainstService = (
+    // 2021-09-15T23:00:00.000Z format
+    productStartDate: string,
+    // 2021-09-15T23:00:00.000Z format
+    productEndDate: string,
+    // 05/04/2020 format
+    serviceStartDate: string,
+    // 05/04/2020 format
+    serviceEndDate: string,
+): boolean => {
+    const momentProductStartDate = moment(productStartDate).valueOf();
+    const momentProductEndDate = moment(productEndDate).valueOf();
+    const momentServiceStartDate = moment(serviceStartDate, 'DD/MM/YYYY').valueOf();
+    const momentServiceEndDate = moment(serviceEndDate, 'DD/MM/YYYY').valueOf();
 
-    return { props: { services: services, myFaresEnabled } };
+    if (momentProductEndDate < momentServiceStartDate || momentServiceEndDate < momentProductStartDate) {
+        return false;
+    }
+    return true;
+};
+
+export const matchProductsToServices = (
+    services: MyFaresService[],
+    products: MyFaresProduct[],
+): MyFaresServiceWithProducts[] =>
+    services.map((service) => {
+        const productsToAdd: MyFaresProduct[] = [];
+        products.forEach((product) => {
```

Comment:
It's going to be much more efficient to do this the other way round - iterate over the services first, and then the products, as otherwise you are iterating over the large products list several times. I'll have a stab at changing it